pyproject.toml

```
@@ -11,7 +11,7 @@ authors = [{name = "Hynek Schlawack", email = "hs@ox.cx"}]
 dynamic = ["version", "description"]
 requires-python = ">=3.7"
 # Keep in-sync with testenv:oldest!
-dependencies = ["prometheus_client >= 0.8.0", "wrapt"]
+dependencies = ["prometheus_client >= 0.8.0", "wrapt", "typing_extensions"]
```

If I read https://github.com/python/typing/commit/4ba98e8c348cd0b2e9b8cc67e6b99eb19a0ed433#diff-6e5e49effcec8f8c76f9823331948755c9cf6925872981dbb44a3c4f95f0b3f3 correctly, we need a at least 3.10.0.

```suggestion
dependencies = ["prometheus_client >= 0.8.0", "wrapt", "typing_extensions >= 3.10.0"]
```

---

Not sure what do you mean, we need `typing_extensions` always? It works on 3.9, I'm using it.

---

this is not a condition, this is the lowest version of typing_extensions that contains ParamSpec :D

---

Ahh I see ;)